WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Details/OrderDetailsDataSourceTests.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import XCTest
+
+import Yosemite
+
+import protocol Storage.StorageManagerType
+import protocol Storage.StorageType
+
+@testable import WooCommerce
+
+/// Test cases for `OrderDetailsDataSourceTests`
+///
+final class OrderDetailsDataSourceTests: XCTestCase {
+
+    private typealias Title = OrderDetailsDataSource.Title
+
+    private var storageManager: StorageManagerType!
+
+    private var storage: StorageType {
+        storageManager.viewStorage
+    }
+
+    override func setUp() {
+        super.setUp()
+        storageManager = MockupStorageManager()
+    }
+
+    override func tearDown() {
+        storageManager = nil
+        super.tearDown()
+    }
+
+    func testPaymentSectionIsShownRightAfterTheProductsAndRefundedProductsSections() {
+        // Given
+        let order = makeOrder()
+
+        insert(refund: makeRefund(orderID: order.orderID, siteID: order.siteID))
+
+        let dataSource = OrderDetailsDataSource(order: order, storageManager: storageManager)
+        dataSource.configureResultsControllers { }
+
+        // When
+        dataSource.reloadSections()
+
+        // Then
+        let actualTitles = dataSource.sections.map(\.title)
```

Comment:
ah interesting using the `title` for comparison, since each section isn't an enum

Reply:
It's not the best but it was the easiest way I could find that works. 😄